CoffeeMachineSimulator/CoffeeMachineSimulator.Services/Services/CoffeeService.cs

```diff
@@ -19,13 +19,18 @@ public CoffeeService()
         public void AddCoffee(CoffeeModel coffeeToAdd)
         {
             if (coffeeToAdd == null) throw new Exception("You should not add null entries!");
-
-            coffeeModels.Add(coffeeToAdd);
+            
+            if (coffeeModels.Any(i => i == coffeeToAdd)) throw new Exception("Already exists an entry with the same Id");
+            if (!(coffeeToAdd.Id == Guid.Empty || coffeeToAdd.Name == null || coffeeToAdd.Price <= 0.0f || coffeeModels.Any(i => i.Name == coffeeToAdd.Name) || coffeeModels.Any(j => j.Id == coffeeToAdd.Id) || coffeeModels.Any(k => k.Price == coffeeToAdd.Price)))
+               
+             coffeeModels.Add(coffeeToAdd);
         }
 
         public void DeleteCoffee(Guid coffeeId)
         {
-            throw new NotImplementedException();
+            if(coffeeId == Guid.Empty) throw new Exception("The Id given is empty");
+            coffeeModels.Remove(coffeeModels.First(i => i.Id == coffeeId));
```

Comment:
please add a check to see if the cofee already exists and after that, try to delete it